src/fileformats/ocd_georef_fields.cpp

```diff
@@ -0,0 +1,457 @@
+	        && lhs.r == rhs.r;
+}
+
+void OcdGeorefFields::setupGeoref(Georeferencing& georef,
+                                  const std::function<void (const QString&)>& warning_handler) const
+{
+	if (m >= 0)
+		georef.setScaleDenominator(m);
+
+	if (qIsFinite(a) && qAbs(a) >= 0.01)
+		georef.setGrivation(a);
+
+	if (r)
+		applyGridAndZone(georef, i, warning_handler);
+
+	QPointF proj_ref_point(x, y);
+	georef.setProjectedRefPoint(proj_ref_point, false);
+}
+
+OcdGeorefFields OcdGeorefFields::fromGeoref(const Georeferencing& georef,
+                                            const std::function<void (const QString&)>& warning_handler)
+{
+	OcdGeorefFields retval;
+
+	auto crs_id_string = georef.getProjectedCRSId();
+	auto crs_param_strings = georef.getProjectedCRSParameters();
+	auto report_warning = [&]() {
+		QStringList params;
+		std::for_each(begin(crs_param_strings), end(crs_param_strings),
+		              [&](const QString& s) { params.append(s); });
+		warning_handler(::OpenOrienteering::OcdFileExport::tr(
+		                    "Could not translate coordinate reference system '%1:%2'.")
+		                .arg(crs_id_string, params.join(QLatin1Char(','))));
+	};
+
+	// translate CRS string into a MapperCrs value
+	auto entry_it = std::find_if(std::begin(crs_string_map),
+	                             std::end(crs_string_map),
+	                             [&crs_id_string](auto e) { return QLatin1String(e.value) == crs_id_string; });
+	if (entry_it == std::end(crs_string_map))
+	{
+		report_warning();
+		return retval;
+	}
+	const MapperCrs crs_id = entry_it->key;
+
+	int crs_param = 0;
+
+	if (crs_id != MapperCrs::Local)
+	{
+		// analyze CRS parameters
+		if (crs_param_strings.empty())
+		{
+			report_warning();
+			return retval;
+		}
+
+		bool param_conversion_ok;
+		auto param_string = crs_param_strings[0];
+		switch (crs_id)
+		{
+		case MapperCrs::Utm:
+			if (param_string.endsWith(QLatin1String(" S")))
+			{
+				param_string.remove(QLatin1String(" S"));
+				param_string.prepend(QLatin1String("-"));
+			}
+			else
+			{
+				param_string.remove(QLatin1String(" N"));
+			}
+			// fallthrough
+
+		default:
+			crs_param = param_string.toInt(&param_conversion_ok);
+		}
+
+		if (!param_conversion_ok)
+		{
+			report_warning();
+			return retval;
+		}
+	}
+
+	auto result = translateMapperToOcd(crs_id, crs_param);
+
+	if (result.ocd_grid_id == OcdGrid::Invalid)
+	{
+		report_warning();
+		return retval;
+	}
+
+	// similar to Mapper, OCD has multiple codes for one CRS
+	// for now we work with the first match
+	retval.a = georef.getGrivation();
+	retval.i = static_cast<int>(result.ocd_grid_id) * 1000 + qAbs(result.ocd_zone_id);
+	// propagate signum which we carry in ocad_zone_id
+	retval.i *= result.ocd_zone_id < 0 ? -1 : 1;
+	retval.m = georef.getScaleDenominator();
+	const QPointF offset(georef.getProjectedRefPoint());
```

Comment:
You need the projected coordinates at 0,0. The existing code uses `georef.toProjectedCoords(MapCoord{})`.

Reply:
Understood. Thanks for the hint!